SWF Editor - Issues on state selection & tooltip overlapping toolbox

romartin opened the pull request:
Hey @LuboTerifaj @hasys 

**See :**
https://issues.redhat.com/browse/KOGITO-8288 
https://issues.redhat.com/browse/KOGITO-8947

**Review actions:**
1.- Check issues in app edit mode (Download the SWF WAR bundle in edit mode at [1]  )
2.- Verify nothing gets broken in SWF VSCode Extension (readonly)

[1] https://drive.google.com/file/d/1ZSkhcyPqAIJrRygI7He_nEta-vtTy2ad/view?usp=sharing
 
Thanks!

hasys commented:
Hi @romartin,

thank you for PR! 

Looks like there is a random test fail:

```
  1 failing

  1) Serverless workflow editor - Basic operations tests
       "after each" hook for "Opens, edits and saves the *.sw.json file":
     StaleElementReferenceError: stale element reference: element is not attached to the page document
  (Session info: chrome=102.0.5005.196)
      at Object.throwDecodedError (/home/runner/work/kie-tools/kie-tools/node_modules/.pnpm/selenium-webdriver@4.8.1_patch_hash=lbqefch5nrzt5atgrk2dnw5phe/node_modules/selenium-webdriver/lib/error.js:524:15)
      at parseHttpResponse (/home/runner/work/kie-tools/kie-tools/node_modules/.pnpm/selenium-webdriver@4.8.1_patch_hash=lbqefch5nrzt5atgrk2dnw5phe/node_modules/selenium-webdriver/lib/http.js:587:13)
      at Executor.execute (/home/runner/work/kie-tools/kie-tools/node_modules/.pnpm/selenium-webdriver@4.8.1_patch_hash=lbqefch5nrzt5atgrk2dnw5phe/node_modules/selenium-webdriver/lib/http.js:515:28)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async Driver.execute (/home/runner/work/kie-tools/kie-tools/node_modules/.pnpm/selenium-webdriver@4.8.1_patch_hash=lbqefch5nrzt5atgrk2dnw5phe/node_modules/selenium-webdriver/lib/webdriver.js:745:17)
      at async toWireValue (/home/runner/work/kie-tools/kie-tools/node_modules/.pnpm/selenium-webdriver@4.8.1_patch_hash=lbqefch5nrzt5atgrk2dnw5phe/node_modules/selenium-webdriver/lib/webdriver.js:145:15)
      at async /home/runner/work/kie-tools/kie-tools/node_modules/.pnpm/selenium-webdriver@4.8.1_patch_hash=lbqefch5nrzt5atgrk2dnw5phe/node_modules/selenium-webdriver/lib/webdriver.js:195:16
      at async forEachKey (/home/runner/work/kie-tools/kie-tools/node_modules/.pnpm/selenium-webdriver@4.8.1_patch_hash=lbqefch5nrzt5atgrk2dnw5phe/node_modules/selenium-webdriver/lib/webdriver.js:189:9)
      at async convertKeys (/home/runner/work/kie-tools/kie-tools/node_modules/.pnpm/selenium-webdriver@4.8.1_patch_hash=lbqefch5nrzt5atgrk2dnw5phe/node_modules/selenium-webdriver/lib/webdriver.js:194:3)
      at async Driver.execute (/home/runner/work/kie-tools/kie-tools/node_modules/.pnpm/selenium-webdriver@4.8.1_patch_hash=lbqefch5nrzt5atgrk2dnw5phe/node_modules/selenium-webdriver/lib/webdriver.js:743:22)
```

To restart the build you need to push empty commit to the branch. To commit an empty PR you can execute this  command:
```
git commit --allow-empty -m "Empty commit - CI Run"
```

Going to check locally meanwhile.

hasys commented on packages/serverless-workflow-diagram-editor/sw-editor/sw-editor-client/src/main/java/org/kie/workbench/common/stunner/sw/client/shapes/icons/CornerIcon.java:
```diff
@@ -60,16 +58,11 @@ public CornerIcon(String icon, Point2D position, String tooltip) {
         });
-        mouseClickHandler = border.addNodeMouseClickHandler(
-                event -> this.getParent().asGroup().getChildren().get(0).fireEvent(event)
-        );
```
Maybe we can change from Goup to Path or getShape instead, it will be clearer and click should still work.

romartin replied:
done , reverted click handler

romartin commented:
Hey @hasys 

Thanks for the review. I rebased from main again, so test failure should gone, also I added again the click handler for the button.

Thanks!

romartin commented:
Thanks a lot @hasys ! Merged, builds re-running

romartin commented:
Hey @LuboTerifaj @hasys 

I updated the link the WAR bundle in the description above, including latest changes from this PR review. So WAR ready to test again!

Thanks!

hasys reviewed (approved):
Hello @romartin,

great job, works good now.

Thank you!

LuboTerifaj reviewed (changes requested):
Hello @romartin ,

I found the following issue:

- If you click directly on the state's icon and keep hovering over it, the tooltip is shown behind toolbox.

Thanks!

romartin commented:
Thanks @LuboTerifaj , good catch! 

Fixed and ready for review again, I updated the link for the WAR file to test in the description of this PR.

Thanks!

romartin commented:
rebased

LuboTerifaj reviewed (approved):
Hi @romartin ,

It looks good to me now.

Thanks for the fix!

LuboTerifaj commented:
Hello @romartin , please consider to include at least one of the jiras' ids to squashed commit message. Thanks!
